[E2E] Optimize looping logic in tests for dashboard date filters

nemanjaglumac opened the pull request:
### Status
PENDING REVIEW

### What does this PR accomplish?
- Optimizes the looping logic in dashboard filters date tests (both GUI and SQL)
- Removes checks for ALL filter types when they are set as the default filter, picks one and makes sure it works

 
This reduces the total running time:
- `dashboard-filters-date` from more than two minutes to about 35 seconds
- `dashboard-filters-sql-date` from more than two minutes to about 45 seconds

In total, we should save between between two and three minutes in total running time just by optimizing date filters in this group.

nemanjaglumac committed:
Optimize looping logic

- This will connect all filter types at once
- We're reducing the number of checks for the default filter value to just one

iethree reviewed (approved):
This looks great to me. Seems to test the same codepaths in a couple bigger tests which saves a lot of time.

One minor concern is that if/when this test does fail, it will be a bit harder to figure out what is broken than with individual tests - but I think the performance tradeoff is worth it here.

nemanjaglumac committed:
Optimize looping logic for SQL dashboard date filters

- This will connect all filter types at once
- We're reducing the number of checks for the default filter value to just one

nemanjaglumac commented:
> One minor concern is that if/when this test does fail, it will be a bit harder to figure out what is broken than with individual tests - but I think the performance tradeoff is worth it here.

I hear you.
Does adding logs in Cypress sidebar like this help?
![image](https://user-images.githubusercontent.com/31325167/181359415-2a3db408-ac89-49c8-b333-cc89105ffc30.png)

nemanjaglumac committed:
Add logs to GUI dashboard date filters

nemanjaglumac commented:
Note: I'll manually backport the whole batch of PRs.
